modules/microshift-providing-sos-report-support.adoc

```diff
@@ -0,0 +1,28 @@
+// Module included in the following assemblies:
+
+// * microshift_support/microshift-sos-report
+
+:_content-type: PROCEDURE
+[id="providing-microshift-sos-report-support_{context}"]
+= Providing a {product-title} sos report to Red Hat support
+
+Red Hat support most commonly looks at an `sos` report when analyzing a service request. The `sos` report provides them with a standardized way to collect diagnostic information, and ensures you are not repeatedly asked for data output.
+
+.Procedure
+
+. To transfer your `sos` report to Red Hat immediatly after generating, run the following command:
++
+[source,terminal]
+----
+$ sudo report --upload
```

Comment:
```suggestion
$ sudo sos report --upload
```

We should verify that this is how support wants this done.

Reply:
@vwalek can you help with this procedure? Do we usually tell users to generate a report and then upload it, or is uploading it when the report is generated normal?

Reply:
@dhellmann yes, we will first collect the sosreport and use the redhat attachment tool to upload that. I haven't seen the option to `--upload`, but I will check it on my lab.

Reply:
Hi. looking at the definition of the upload, the upload should be used if you have specific server you want to push it. 
```
  --upload              Upload archive to a policy-default location
  --upload-url UPLOAD_URL
                        Upload the archive to specified server
  --upload-directory UPLOAD_DIRECTORY
                        Specify upload directory for archive
  --upload-user UPLOAD_USER
                        Username to authenticate to server with
  --upload-pass UPLOAD_PASS
                        Password to authenticate to server with
  --upload-method {auto,put,post}
                        HTTP method to use for uploading
  --upload-no-ssl-verify
                        Disable SSL verification for upload url
  --upload-protocol {auto,https,ftp,sftp}
                        Manually specify the upload protocol
```

I am not sure if we want to document that you can push it directly to some location. 

I would leave it optional, like you can push it to your own server using SFTP, SCP or HTTP protocols.

Usually, the sosreport is collected to the bastion host and use of the redhat attachment tool or run the tool directly from the RHEL system.

I don't think the `--upload` should be part of the sos collection docs.

When running the `sos report` the system will ask you the ticket number to upload it directly, but that's optional.

Reply:
Thanks !!

Reply:
Thank you! We will leave those out.